src/libs/actions/OnyxUpdateManager.ts

```diff
@@ -41,11 +41,18 @@ export default () => {
     Onyx.connect({
         key: ONYXKEYS.ONYX_UPDATES_FROM_SERVER,
         callback: (value) => {
-            // When the OpenApp command hasn't finished yet, we should not process any updates.
-            if (!value || isLoadingApp) {
-                if (isLoadingApp) {
-                    console.debug(`[OnyxUpdateManager] Ignoring Onyx updates while OpenApp hans't finished yet.`);
-                }
+            // When there's no value, there's nothing to process, so let's return early.
+            if (!value) {
+                return;
+            }
+            // If isLoadingApp is positive it means that OpenApp command hasn't finished yet, and in that case
+            // we don't have base state of the app (reports, policies, etc) setup. If we apply this update,
+            // we'll only have them overriten by the openApp response. So let's skip it and return.
+            if (isLoadingApp) {
+                // When ONYX_UPDATES_FROM_SERVER is set, we pause the queue. Let's unpause
+                // it so the app is not stuck forever without processing requests.
+                SequentialQueue.unpause();
```

Comment:
I wonder if there is a better place to put this. What if you put it in the callback for `ONYXKEYS.IS_LOADING_APP` when `value` goes from `true` to `false`? Then we know it's safe to unpause the queue (because that's when we know that `OpenApp` is finished).

With it being down here, it seems a little too brittle like... there is no real guarantee that this spot means `OpenApp` is finished yet and it's safe to unpause the queue.

Reply:
Long discussion on Slack here to address this https://expensify.slack.com/archives/C06SCNUBB28/p1712948023511979